File: packages/python-client-sdk/src/sdk/models/operations/query_subscriptions_all.py
import dataclasses
import dateutil.parser
from dataclasses_json import dataclass_json
from datetime import datetime
from enum import Enum
from marshmallow import fields
from sdk import utils
from typing import Any, Optional


@dataclasses.dataclass
class QuerySubscriptionsAllQueryParams:
    product_id: str = dataclasses.field(metadata={'query_param': { 'field_name': 'productId', 'style': 'form', 'explode': True }})
    

@dataclasses.dataclass
class QuerySubscriptionsAllRequest:
    query_params: QuerySubscriptionsAllQueryParams = dataclasses.field()
    

@dataclass_json
@dataclasses.dataclass
class QuerySubscriptionsAllDefaultApplicationJSONIssues:
    message: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('message') }})
    

@dataclass_json
@dataclasses.dataclass
class QuerySubscriptionsAllDefaultApplicationJSON:
    code: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('code') }})
    message: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('message') }})
    issues: Optional[list[QuerySubscriptionsAllDefaultApplicationJSONIssues]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('issues') }})
    
class QuerySubscriptionsAll200ApplicationJSONSeatingConfigSeatingStrategyNameEnum(str, Enum):
    MONTHLY_ACTIVE_USER = "monthly_active_user"
    FIRST_COME_FIRST_SERVED = "first_come_first_served"


@dataclass_json
@dataclasses.dataclass
class QuerySubscriptionsAll200ApplicationJSONSeatingConfig:
    default_low_seat_warning_level_percent: float = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('default_low_seat_warning_level_percent') }})
    limited_overflow_seating_enabled: bool = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('limited_overflow_seating_enabled') }})
    low_seat_warning_level_pct: float = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('low_seat_warning_level_pct') }})
    owner_id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('owner_id') }})
    seating_strategy_name: QuerySubscriptionsAll200ApplicationJSONSeatingConfigSeatingStrategyNameEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('seating_strategy_name') }})
    default_seat_expiry_in_days: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('default_seat_expiry_in_days') }})
    seat_reservation_expiry_in_days: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('seat_reservation_expiry_in_days') }})
    
class QuerySubscriptionsAll200ApplicationJSONStateEnum(str, Enum):
    PURCHASED = "purchased"
    ACTIVE = "active"
    SUSPENDED = "suspended"
    CANCELED = "canceled"


@dataclass_json
@dataclasses.dataclass
class QuerySubscriptionsAll200ApplicationJSON:
    admin_email: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('admin_email') }})
    admin_name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('admin_name') }})
    admin_role_name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('admin_role_name') }})
    created_utc: datetime = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('created_utc'), 'encoder': utils.datetimeisoformat(False), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    is_being_configured: bool = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('is_being_configured') }})
    is_free_trial: bool = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('is_free_trial') }})
    is_setup_complete: bool = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('is_setup_complete') }})
    is_test_subscription: bool = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('is_test_subscription') }})
    offer_id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('offer_id') }})
    plan_id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('plan_id') }})
    product_id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('product_id') }})
    seating_config: QuerySubscriptionsAll200ApplicationJSONSeatingConfig = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('seatingConfig') }})
    state: QuerySubscriptionsAll200ApplicationJSONStateEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('state') }})
    state_last_updated_utc: datetime = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('state_last_updated_utc'), 'encoder': utils.datetimeisoformat(False), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    subscription_name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('subscription_name') }})
    tenant_id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('tenant_id') }})
    tenant_name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('tenant_name') }})
    total_seats: int = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('total_seats') }})
    user_role_name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('user_role_name') }})
    management_urls: Optional[Any] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('management_urls') }})
    source_subscription: Optional[Any] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('source_subscription') }})
    subscriber_info: Optional[Any] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('subscriber_info') }})
    

@dataclasses.dataclass
class QuerySubscriptionsAllResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    query_subscriptions_all_200_application_json_objects: Optional[list[QuerySubscriptionsAll200ApplicationJSON]] = dataclasses.field(default=None)
    query_subscriptions_all_default_application_json_object: Optional[QuerySubscriptionsAllDefaultApplicationJSON] = dataclasses.field(default=None)
    

File: packages/ts-client-sdk/src/sdk/models/operations/querysubscriptionsall.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";


export class QuerySubscriptionsAllQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=productId" })
  productId: string;
}

export class QuerySubscriptionsAllRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: QuerySubscriptionsAllQueryParams;
}

export class QuerySubscriptionsAllDefaultApplicationJSONIssues extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=message" })
  message: string;
}

export class QuerySubscriptionsAllDefaultApplicationJSON extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=code" })
  code: string;

  @SpeakeasyMetadata({ data: "json, name=issues", elemType: QuerySubscriptionsAllDefaultApplicationJSONIssues })
  issues?: QuerySubscriptionsAllDefaultApplicationJSONIssues[];

  @SpeakeasyMetadata({ data: "json, name=message" })
  message: string;
}
export enum QuerySubscriptionsAll200ApplicationJSONSeatingConfigSeatingStrategyNameEnum {
    MonthlyActiveUser = "monthly_active_user",
    FirstComeFirstServed = "first_come_first_served"
}

export class QuerySubscriptionsAll200ApplicationJSONSeatingConfig extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=default_low_seat_warning_level_percent" })
  defaultLowSeatWarningLevelPercent: number;

  @SpeakeasyMetadata({ data: "json, name=default_seat_expiry_in_days" })
  defaultSeatExpiryInDays?: number;

  @SpeakeasyMetadata({ data: "json, name=limited_overflow_seating_enabled" })
  limitedOverflowSeatingEnabled: boolean;

  @SpeakeasyMetadata({ data: "json, name=low_seat_warning_level_pct" })
  lowSeatWarningLevelPct: number;

  @SpeakeasyMetadata({ data: "json, name=owner_id" })
  ownerId: string;

  @SpeakeasyMetadata({ data: "json, name=seat_reservation_expiry_in_days" })
  seatReservationExpiryInDays?: number;

  @SpeakeasyMetadata({ data: "json, name=seating_strategy_name" })
  seatingStrategyName: QuerySubscriptionsAll200ApplicationJSONSeatingConfigSeatingStrategyNameEnum;
}
export enum QuerySubscriptionsAll200ApplicationJSONStateEnum {
    Purchased = "purchased",
    Active = "active",
    Suspended = "suspended",
    Canceled = "canceled"
}

export class QuerySubscriptionsAll200ApplicationJSON extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=admin_email" })
  adminEmail: string;

  @SpeakeasyMetadata({ data: "json, name=admin_name" })
  adminName: string;

  @SpeakeasyMetadata({ data: "json, name=admin_role_name" })
  adminRoleName: string;

  @SpeakeasyMetadata({ data: "json, name=created_utc" })
  createdUtc: Date;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "json, name=is_being_configured" })
  isBeingConfigured: boolean;

  @SpeakeasyMetadata({ data: "json, name=is_free_trial" })
  isFreeTrial: boolean;

  @SpeakeasyMetadata({ data: "json, name=is_setup_complete" })
  isSetupComplete: boolean;

  @SpeakeasyMetadata({ data: "json, name=is_test_subscription" })
  isTestSubscription: boolean;

  @SpeakeasyMetadata({ data: "json, name=management_urls" })
  managementUrls?: any;

  @SpeakeasyMetadata({ data: "json, name=offer_id" })
  offerId: string;

  @SpeakeasyMetadata({ data: "json, name=plan_id" })
  planId: string;

  @SpeakeasyMetadata({ data: "json, name=product_id" })
  productId: string;

  @SpeakeasyMetadata({ data: "json, name=seatingConfig" })
  seatingConfig: QuerySubscriptionsAll200ApplicationJSONSeatingConfig;

  @SpeakeasyMetadata({ data: "json, name=source_subscription" })
  sourceSubscription?: any;

  @SpeakeasyMetadata({ data: "json, name=state" })
  state: QuerySubscriptionsAll200ApplicationJSONStateEnum;

  @SpeakeasyMetadata({ data: "json, name=state_last_updated_utc" })
  stateLastUpdatedUtc: Date;

  @SpeakeasyMetadata({ data: "json, name=subscriber_info" })
  subscriberInfo?: any;

  @SpeakeasyMetadata({ data: "json, name=subscription_name" })
  subscriptionName: string;

  @SpeakeasyMetadata({ data: "json, name=tenant_id" })
  tenantId: string;

  @SpeakeasyMetadata({ data: "json, name=tenant_name" })
  tenantName: string;

  @SpeakeasyMetadata({ data: "json, name=total_seats" })
  totalSeats: number;

  @SpeakeasyMetadata({ data: "json, name=user_role_name" })
  userRoleName: string;
}

export class QuerySubscriptionsAllResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata({ elemType: QuerySubscriptionsAll200ApplicationJSON })
  querySubscriptionsAll200ApplicationJSONObjects?: QuerySubscriptionsAll200ApplicationJSON[];

  @SpeakeasyMetadata()
  querySubscriptionsAllDefaultApplicationJSONObject?: QuerySubscriptionsAllDefaultApplicationJSON;
}